DataStructures/MinMaxHeap.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures
+{
+    /// <summary>
+    /// This class implements min-max heap.
+    /// It provides functionality of both min-heap and max-heap with the same time complexity.
+    /// Therefore it provides constant time retrieval and logarithmic time removal
+    /// of both the minimum and maximum elements in it.
+    /// </summary>
+    /// <typeparam name="T">Generic type.</typeparam>
+    public class MinMaxHeap<T>
+    {
+        private readonly List<T> heap = new List<T>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MinMaxHeap{T}"/> class that contains
+        /// elements copied from a specified enumerable collection and that uses a specified comparer.
+        /// </summary>
+        /// <param name="collection">The enumerable collection to be copied.</param>
+        /// <param name="comparer">The default comparer to use for comparing objects.</param>
+        public MinMaxHeap(IEnumerable<T>? collection = null, IComparer<T>? comparer = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5421646)